src/main/java/me/chrr/scribble/mixin/BookEditScreenMixin.java

```diff
@@ -274,50 +315,92 @@ public void init(PlayerEntity player, ItemStack itemStack, Hand hand, CallbackIn
         }
     }
 
+    @Unique
+    private void changeActiveColor(@NotNull Formatting color) {
+        if (this.activeColor == color) {
+            return;
+        }
+
+        Command command = new ActionCommand<>(this, () -> {
+            activeColor = color;
+            invalidateFormattingButtons();
+            getRichSelectionManager().applyColorForSelection(color);
+        });
+        commandManager.execute(command);
+    }
+
+    @Unique
+    public void toggleActiveModifier(Formatting modifier, boolean toggled) {
+        Command command = new ActionCommand<>(this, () -> {
+            if (toggled) {
+                activeModifiers.add(modifier);
+            } else {
+                activeModifiers.remove(modifier);
+            }
+            invalidateFormattingButtons();
+
+            // ToDo replace with manager.applyModifiersForSelection(activeModifiers) call
+            //  to have the single state of truth for activeModifiers.
+            getRichSelectionManager().toggleModifierForSelection(modifier, toggled);
+        });
+        commandManager.execute(command);
+    }
+
     @Inject(method = "init", at = @At(value = "HEAD"))
     private void initScreen(CallbackInfo ci) {
         initButtons();
-
-        // We need to update the states of all the buttons again.
-        this.getRichSelectionManager().updateSelectionFormatting();
     }
 
     @Inject(method = "updateButtons", at = @At(value = "HEAD"))
     private void updateButtons(CallbackInfo ci) {
-        this.boldButton.visible = !this.signing;
-        this.italicButton.visible = !this.signing;
-        this.underlineButton.visible = !this.signing;
-        this.strikethroughButton.visible = !this.signing;
-        this.obfuscatedButton.visible = !this.signing;
+        Optional.ofNullable(boldButton).ifPresent(button -> button.visible = !this.signing);
+        Optional.ofNullable(italicButton).ifPresent(button -> button.visible = !this.signing);
+        Optional.ofNullable(underlineButton).ifPresent(button -> button.visible = !this.signing);
+        Optional.ofNullable(strikethroughButton).ifPresent(button -> button.visible = !this.signing);
+        Optional.ofNullable(obfuscatedButton).ifPresent(button -> button.visible = !this.signing);
```

Comment:
All these `ofNullable` checks really makes the code messy. As far as I know, the buttons should always be initialized before we do something with them, is there any reason to check every time?

Reply:
The `RichSelectionManager::updateSelectionFormatting()` is no longer needed, because `RichSelectionManager`do not have `activeColor` and `activeModifiers` states. They are now [stored](https://github.com/wiskiw/ScribblePlusMC/blob/a93f3e4f8dd4c8acb19bdefc712b30d07e4878bb/src/main/java/me/chrr/scribble/mixin/BookEditScreenMixin.java#L129C29-L129C44) in BookEditScreen now. But to keep these values up to date in cases, when the user moves cursor `RichSelectionManager` we have a callback - `StateCallback::onCursorFormattingChanged`. 

From an abstract point of view, when we create RichSelectionManager and pass this callback we don't know when it could be called. And the implementation of the callback should be safe to be executed at any moment.

In the real example, in BookEditScreen we update the buttons every time when this callback is executed. But since the buttons initialized after the RichSelectionManager, we need to have a null check for every button/swatches that are used in the callback. 

Hope it's cleaner now. I agree it makes code more messy, but I guess it's a good save measure to prevent runtime exception (I faced it during development), since we do not initialize buttons in `RichSelectionManager::init()`

Reply:
Alright, makes sense, we'll live with it then.